x-pack/plugins/timelines/public/components/t_grid/helpers.tsx

```diff
@@ -388,3 +406,66 @@ export const getDefaultViewSelection = ({
 
 /** This local storage key stores the `Grid / Event rendered view` selection */
 export const ALERTS_TABLE_VIEW_SELECTION_KEY = 'securitySolution.alerts.table.view-selection';
+
+const buildISQueryMatch = ({
+  browserFields,
+  field,
+  isFieldTypeNested,
+  value,
+}: {
+  browserFields: BrowserFields;
+  field: string;
+  isFieldTypeNested: boolean;
+  value: string | number;
+}): string => {
+  if (isFieldTypeNested) {
+    return convertNestedFieldToQuery(field, value, browserFields);
+  } else if (checkIfFieldTypeIsDate(field, browserFields)) {
+    return convertDateFieldToQuery(field, value);
+  } else {
+    return `${field} : ${isNumber(value) ? value : escapeQueryValue(value)}`;
+  }
+};
+
+const buildEXISTSQueryMatch = ({
+  browserFields,
+  field,
+  isFieldTypeNested,
+}: {
+  browserFields: BrowserFields;
+  field: string;
+  isFieldTypeNested: boolean;
+}): string => {
+  return isFieldTypeNested
+    ? convertNestedFieldToExistQuery(field, browserFields)
+    : `${field} ${EXISTS_OPERATOR}`;
+};
+
+const buildISONEOFQueryMatch = ({
+  browserFields,
+  field,
+  isFieldTypeNested,
+  value,
+}: {
+  browserFields: BrowserFields;
+  field: string;
+  isFieldTypeNested: boolean;
+  value: Array<string | number>;
+}): string => {
+  const formattedValues = value.map((item) => (isNumber(item) ? Number(item) : item));
+
+  return `${field} : ${JSON.stringify(formattedValues)
+    .replace(/^\[/, '(')
+    .replace(/\]$/g, ')')
+    .replace(/,/g, ' OR ')}`;
+};
+
+function isStringOrNumberArray(
+  val: string | number | Array<string | number>
+): val is Array<string | number> {
+  return (
+    typeof val !== 'string' &&
```

Comment:
Same thing here, you can just check first if `Array.isArray(value) instead of the val string and number checks` Then you can re-use this function above.

Maybe an over-engineered solution could be:

export IsArrayOfOptions = 'string' | 'number' | 'object' .... etc

export const isArrayOf = (value: unknown, options: IsArrayOfOptions[]) =>  Array.isArray(value) && options.some((typeOption) => typeof value[0] === typeOption);

isArrayOf(val, ['number', 'string']);